### ISSUE TITLE
EACCES: permission denied wbo should check that is has the right permissions in the board persistence folder

### ISSUE THREAD

**Issue by NPCGundlach** [lovasoa/whitebophir#164]
Apparently boards are not being saved on the latest docker release. In the terminal I get the following error:  
`2021-01-27T15:09:23.830Z        board saving error      {"err":"Error: EACCES: permission denied, open '/opt/app/server-data/board-adsads.json.2021-01-27T150923.830Z.bak'","tmp_file":"/opt/app/server-data/board-adsads.json.2021-01-27T150923.830Z.bak"}`

Happens on both a Ubuntu 16.04 and Debian 10 machine

Reproduction steps:
From the terminal:
`user@host:~$ sudo docker run -it --publish 5001:8080 --volume "$(pwd)/wbo-boards:/opt/app/server-data" lovasoa/wbo:latest `  
Open a board by connecting to host:5001  
Draw something  

The error is going to be written into the terminal. The folder `~/wbo-boards/` stays empty. Reloading the board shows an empty board.

**Comment by NPCGundlach** [comment]
I've gone through the past releases and the issue seems to happen in the docker containers `1.7.0`, too. The docker container running under `1.6.5` works fine.

**Comment by lovasoa** [comment]
This doesn't sound like a bug in WBO, but rather a permission issue in your local setup. Can you run from inside the container (with docker exec for instance):

```
ls -lah /opt/app/server-data/
```

and

```
touch /opt/app/server-data/test.json
```

The dockerfile was [updated](https://github.com/lovasoa/whitebophir/commit/ce5ca084b9b9bc73341123695604d6fc9a4cc946#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557) recently to avoid running as root inside the container. If your board files belong to root, they won't be accessible, and you should update the permissions on them.

**Comment by NPCGundlach** [comment]
Sure, this is the output of executing the commands within the containers:

```
user@host:~$ sudo docker container ls
CONTAINER ID   IMAGE                         COMMAND                  CREATED         STATUS         PORTS                      NAMES
c82b337fdff3   lovasoa/wbo:latest            "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:5001->8080/tcp     nostalgic_kepler

user@host:~$ sudo docker container exec c82b ls -lah /opt/app/server-data/
total 16K
drwxr-xr-x    2 root     root        4.0K Jan 27 15:30 .
drwxr-xr-x    1 node     node        4.0K Jan 13 09:04 ..
-rw-r--r--    1 root     root        5.3K Jan 27 15:30 board-adsads.json

user@host:~$ sudo docker container exec c82b touch /opt/app/server-data/test.json
touch: /opt/app/server-data/test.json: Permission denied
```

I'm not overly familiar with Docker. Do I need to manually change permission of the folders in the container? Would I then have to do that every time I restart the container?

**Comment by lovasoa** [comment]
I should have written migration notes detailing the steps to take to avoid this problem. I didn't realize that at the moment.

What you should run (from the same directory as the one in which you launched `docker run ...`) is 

```
sudo chown 1000:1000 -R wbo-boards
```

**Comment by NPCGundlach** [comment]
Ah, that fixed it, thanks! 
Not sure if migration notes would've helped - I don't think I would've read them considering this is the first time I ran wbo on this machine.

**Comment by lovasoa** [comment]
An action point could be for WBO to check that it has read/write permissions on the boards folder when it starts. If it doesn't, it could tell the user about the problem (and the error message may include the chown command to fix it). Are you interested in implementing that, @NPCGundlach ?

**Comment by NPCGundlach** [comment]
I would be interested, but at the moment my work contract states that all of the work I do on company time belongs to the company, so I can't contribute code to Open Source software :/  
And sadly, due to a health issue on my part my time off-work is severely limited, so that I haven't been able to spend much time on writing software as a hobby since starting full time...

**Comment by lovasoa** [comment]
No worries ! Let's keep this issue opened for the moment. I'll highlight the solution in case other people stumble on this thread while searching for the error message.

**Comment by lovasoa** [comment]
I released wbo v1.7.1 which does a complete permission check on startup
